tests/taipy/gui/cli/test_cli.py

```diff
@@ -0,0 +1,155 @@
+
+    _inject_section(
+        _GuiSection,
+        "gui_config",
+        _GuiSection(property_list=list(default_config)),
+        [("configure_gui", _GuiSection._configure)],
+        add_to_unconflicted_sections=True,
+    )
+
+
+def test_gui_service_arguments_hierarchy():
+    # Test default configuration
+    gui = Gui()
+    gui.run(run_server=False)
+    service_config = gui._config.config
+    assert not service_config["allow_unsafe_werkzeug"]
+    assert service_config["async_mode"] == "gevent"
+    assert service_config["change_delay"] is None
+    assert service_config["chart_dark_template"] is None
+    assert service_config["dark_mode"]
+    assert service_config["dark_theme"] is None
+    assert not service_config["debug"]
+    assert not service_config["extended_status"]
+    assert service_config["favicon"] is None
+    assert not service_config["flask_log"]
+    assert service_config["host"] == "127.0.0.1"
+    assert service_config["light_theme"] is None
+    assert service_config["margin"] is None
+    assert service_config["ngrok_token"] == ""
+    assert service_config["notification_duration"] == 3000
+    assert service_config["propagate"]
+    assert service_config["run_browser"]
+    assert not service_config["run_in_thread"]
+    assert not service_config["run_server"]
+    assert not service_config["single_client"]
+    assert not service_config["system_notification"]
+    assert service_config["theme"] is None
+    assert service_config["time_zone"] is None
+    assert service_config["title"] is None
+    assert service_config["upload_folder"] is None
+    assert not service_config["use_arrow"]
+    assert not service_config["use_reloader"]
+    assert service_config["watermark"] == "Taipy inside"
+    assert service_config["webapp_path"] is None
+    assert service_config["port"] == 5000
+    gui.stop()
+
+    # Override default configuration by explicit defined arguments in Gui.run()
+    gui = Gui()
+    gui.run(run_server=False, watermark="", host="my_host", port=5001)
+    service_config = gui._config.config
+    assert service_config["watermark"] == ""
+    assert service_config["host"] == "my_host"
+    assert service_config["port"] == 5001
+    gui.stop()
+
+    # Override Gui.run() arguments by explicit defined arguments in Config.configure_gui()
+    Config.configure_gui(dark_mode=False, host="my_2nd_host", port=5002)
+    gui = Gui()
+    gui.run(run_server=False, watermark="", host="my_host", port=5001)
+    service_config = gui._config.config
+    assert not service_config["dark_mode"]
+    assert service_config["host"] == "my_2nd_host"
+    assert service_config["watermark"] == ""
+    assert service_config["port"] == 5002
+    gui.stop()
+
+    # Override Config.configure_gui() arguments by loading a TOML file with [gui] section
+    toml_config = NamedTemporaryFile(
+        content="""
+[TAIPY]
+
+[gui]
+host = "my_3rd_host"
+port = 5003
+use_reloader = "true:bool"
```

Comment:
I don't understand what that is not simply
`use_reloader = true`

That would be very valid TOML...

Reply:
In taipy-config, we are using the `:bool` to separate a boolean from a string.

This is just how it was implemented.